doc/source/development/debugging_extensions.rst

```diff
@@ -0,0 +1,69 @@
+.. _debugging_c_extensions:
+
+{{ header }}
+
+======================
+Debugging C extensions
+======================
+
+Pandas uses select C extensions for high performance IO operations. In case you need to debug segfaults or general issues with those extensions, the following steps may be helpful. These steps are geared towards using lldb as a debugger, though the steps for gdb will be similar.
+
+First, be sure to compile the extensions with the appropriate flags to generate debug symbols and remove optimizations. This can be achieved as follows:
+
+.. code-block:: sh
+
+   python setup.py build_ext --inplace -j4 --with-debugging-symbols
+
+Using a debugger
+================
+
+You can create a script that hits the extension module you are looking to debug and place it in the project root. Thereafter launch a Python process under lldb:
```

Comment:
is there a reference on lldb (link / how to install?)

Reply:
I have only used on macOS where it comes bundled with the Xcode tools. Not sure about other systems

Reply:
On my ubuntu machine it's not available (while `gdb` is). So I would at least link to some web page about it if we want to recommend it over gdb, I suppose most readers won't know what lldb is (I also didn't). 
it seems you can install it from conda-forge.

Reply:
To be clear I don't recommend one over the other - should just use whatever comes with your build system. 

I'll try to reword

Reply:
Did you settle on a wording here?